Week-4/session-10/callBindApply.js

```diff
@@ -0,0 +1,38 @@
+class Calculator{
+  add(num1,num2){
+      return num1+num2
+  }
+  subtract(num1,num2){
+      return num1-num2
+  }
+  multiply(num1,num2){
+      return num1*num2
+  }
+  divide(num1,num2){
+      return num1/num2
+  }
+}
+
+class ScientificCalculator extends Calculator {
+  square(num){
+      return num**2
+  }
+  cube(num){
+      return num**3
+  }
+  power(num,raise){
+      return num**raise
+  }
+  multiplyByTwo(num) {
+      return num * 2;
+  }
+  powerOfThree(num){
+      return num**3
+  }
+}
+
+let obj = new ScientificCalculator
+let addResult = obj.add.call(obj, 10, 5);
+let subResult = obj.subtract.apply(obj, [10, 5]);
+let multiplyByTwo = obj.multiplyByTwo.bind(obj)
```

Comment:
why to use call ,apply, bind on obj when it's already an instance of the ScientificCalculator , I can directly call obj.add
<img width="715" alt="image" src="https://github.com/pesto-students/p14-harsh-srivastava-deepakk5893/assets/127043221/7a04dcd8-e2b3-4811-a346-3eb401ecd556">
revisit this concept as you are not using it coorectly